lib/src/revset.rs

```diff
@@ -933,12 +936,29 @@ fn parse_builtin_function(
             Ok(RevsetExpression::public_heads())
         }
         "branches" => {
-            expect_no_arguments(name, arguments_pair)?;
-            Ok(RevsetExpression::branches())
+            let mut needle = "".to_owned();
+            if let Some(arg) = expect_one_optional_argument(name, arguments_pair)? {
+                needle = parse_function_argument_to_string(name, arg, state)?;
+            }
```

Comment:
nit: I think it's more Rustic to write this as you have done in the "remote_branches" arm (`let needle = if ...`)

Reply:
or `.map(|arg| ...).transpose()?.unwrap_or_default()`? But, `if ...` woud be easier to read.

Reply:
Yeah, it just felt a little too dense, added an intermediate opt_arg and adopted this.